app/src/main/java/org/mozilla/fenix/shortcut/FirstTimePwaObserver.kt

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shortcut
+
+import androidx.lifecycle.LifecycleOwner
+import androidx.navigation.NavController
+import mozilla.components.browser.session.Session
+import mozilla.components.concept.engine.manifest.WebAppManifest
+import mozilla.components.feature.pwa.WebAppUseCases
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.BrowserFragmentDirections
+import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.utils.Settings
+
+class FirstTimePwaObserver(
+    session: Session?,
+    owner: LifecycleOwner,
+    private val navController: NavController,
+    private val settings: Settings,
+    private val webAppUseCases: WebAppUseCases
+) : Session.Observer {
+
+    init {
+        if (session != null && settings.shouldShowFirstTimePwaFragment) {
+            session.register(this, owner, autoPause = true)
```

Comment:
Any reason we can't just use the LifecycleAwareFeature (rename to `FirstTimePwaFeature` as well)? Seems like you're trying to do the same for it already.

Reply:
I register it here so that I can do the settimgs check you mentioned in the other comment, without having to repeat the check if we use this class elsewhere.

Reply:
I think we want to move away from this style of "setup observers in the init block". I'm quite guilty of doing this myself, but @csadilek had some good points to avoid it.

See: https://github.com/mozilla-mobile/android-components/issues/5449